db/db_common/db_session.go

```diff
@@ -33,9 +35,22 @@ func (s *DatabaseSession) UpdateUsage() {
 	s.LastUsed = time.Now()
 }
 
-func (s *DatabaseSession) Close() error {
+func (s *DatabaseSession) Close(waitForCleanup bool) error {
+	var err error
 	if s.Connection != nil {
-		err := s.Connection.Close()
+		if waitForCleanup {
+			s.Connection.Raw(func(driverConn interface{}) error {
+				conn := driverConn.(*stdlib.Conn)
+				select {
+				case <-time.After(5 * time.Second):
```

Comment:
is 5 secs long enough?
what happens if we timeout? nothing? i.e. we just continue shutting down? That seems desirable

Reply:
for now, we just continue.